File: ExamTest/app/src/main/java/ru/project/examtest/ui/basic_formulas/AdapterViewPager.java
package ru.project.examtest.ui.basic_formulas;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class AdapterViewPager extends FragmentStateAdapter {
    private FragmentActivity fragmentActivity;
    private Fragment[] fragments;

    public AdapterViewPager(
            FragmentActivity fragmentActivity,
            Fragment[] fragments
    ) {
        super(fragmentActivity);
        this.fragmentActivity = fragmentActivity;
        this.fragments = fragments;
    }

    public void setFragmentActivity(FragmentActivity fragmentActivity) {
        this.fragmentActivity = fragmentActivity;
    }

    public void setFragments(Fragment[] fragments) {
        this.fragments = fragments;
    }

    public Fragment[] getFragments() {
        return fragments;
    }

    public FragmentActivity getFragmentActivity() {
        return fragmentActivity;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
            return getFragments()[position];
    }

    @Override
    public int getItemCount() {
        return getFragments().length;
    }
}

File: TestAdapter/app/src/main/java/ru/project/testadapter/AdapterViewPager.kt
package ru.project.testadapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class AdapterViewPager(
    var listFragment: List<Fragment>,
    var fragmentActivity: FragmentActivity
    ): FragmentStateAdapter(fragmentActivity) {

    override fun getItemCount(): Int = listFragment.size

    override fun createFragment(position: Int): Fragment = listFragment[position]
}